docs/specs/authentication-flow.md

```diff
@@ -356,6 +357,104 @@ reauth_flows:
     - authentication: secondary_sms_code
 ```
 
+### Flow selection
+
+By default, authentication flow with name `default` is used. Since there can be multiple flows of same type, we want client to be able to select a flow group to use.
+
+For example, given following authentication flows:
+
+```yaml
+authentication_flow:
+  signup_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  login_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  signup_login_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  reauth_flows:
+    - name: sms_code
+      steps:
+      # ...
+    - name: password
+      steps:
+      # ...
+  promote_flows:
+    - name: password
+      steps:
+      # ...
+    - name: sms_code
+      steps:
+      # ...
+  account_recovery_flows:
+    - name: email
+      steps:
+      # ...
+    - name: sms
+      steps:
+      # ...
+```
+
+Authentication flow groups need to be defined and whitelisted per client using the following configuration:
+
+```yaml
+ui:
+  authentication_flow:
+    groups:
+    - name: oauth_only
```

Comment:
Can `groups[].name = "default` be overridden or reserved?

Reply:
Can be overridden.